3ds/source/configuration.cpp

```diff
@@ -105,6 +123,13 @@ Configuration::Configuration(void)
         }
         mAdditionalExtdataFolders.emplace(strtoull(it.key().c_str(), NULL, 16), u16folders);
     }
+    
+    auto extdata = mJson["extdata_locations"];
+    for (auto it = extdata.begin(); it != extdata.end(); ++it) {
+        u32 key = std::stoul(((std::string) it.key()).c_str(), nullptr, 16);
```

Comment:
This will result in unnecessary copying. Could you get rid of the `.c_str()`?

Reply:
Done.

Reply:
To clarify, I had included the `c_str()` calls because I had noted that they were already in use in stoul calls in the same file (see e.g. lines 111 and 124), and I tend to err on the side of convention when contributing to codebases. Perhaps those should be revised as well?